pkg/udn/primary_network.go

```diff
@@ -0,0 +1,37 @@
+package udn
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubevirt/ipam-extensions/pkg/config"
+)
+
+func FindPrimaryNetwork(ctx context.Context,
+	cli client.Client,
+	namespace string) (*v1.NetworkAttachmentDefinition, error) {
+	nadList := v1.NetworkAttachmentDefinitionList{}
+	if err := cli.List(ctx, &nadList, client.InNamespace(namespace)); err != nil {
```

Comment:
So:
1. Controller runtime client has already an informer caching stuff behing the scenes.
2. That's interesting, but let's consider it if we hit a performance issue (Is not that we are going to have a massive number of NADs per namespace)
3. We can still have multiple NADs with that possible label, is the same scenario, so we either take the first one and not validate or validate, since there is not going to be a massive number of NADs and they are already cache at the client we can validate.

Reply:
> 3. We can still have multiple NADs with that possible label, is the same scenario, so we either take the first one and not validate or validate, since there is not going to be a massive number of NADs and they are already cache at the client we can validate.

What i mean is that for one NS, we can't have multiple NADs with that label, if OVN is the one that makes sure it put it on the first that is created and assert no others are created as primary on that NS (this of course involved new required logic if accepted).
Anyhow it is just food for thought in the future.
And then it gives you [2] for free, but again just nice to have as you said.

I just envision this repo to be allowed to assume stuff that is given, asserted by OVN etc.